xla/stream_executor/cuda/ptx_compiler_helpers.cc

```diff
@@ -56,4 +56,24 @@ absl::Status CreateErrorFromPTXASLog(std::string_view log,
   return absl::OkStatus();
 }
 
+// Warns if the ptxas version should be upgraded.
+// Only prints the warning upon the first invocation.
+void WarnIfBadPtxasVersion(const CudaComputeCapability& cc,
+                           SemanticVersion ptxas) {
+  static absl::once_flag run_once;
+  absl::call_once(run_once, [&] {
+    // nvbug 4826023: Occurs on Hopper+ in CUDA versions 12.x up to and
+    // including CUDA 12.6.2; the earliest ptxas release that corresponds to
+    // CUDA 12.6.3 is ptxas 12.6.85.
+    if (cc.major >= 9 && ptxas >= SemanticVersion{12, 0, 0} &&
+        ptxas < SemanticVersion{12, 6, 85}) {
+      LOG(ERROR)
+          << "*** WARNING *** Invoking PTXAS with version " << ptxas
```

Comment:
Can you make `PTXAS` a std::string_view parameter in the function, so that we can specify wether we use libnvjitlink or libnvptxcompiler for compilation.